frontend/src/App.js

```diff
@@ -1,21 +1,53 @@
-import React, { useState, useEffect } from "react";
+import React, { useState, useEffect, useRef } from "react";
 import Form from "@rjsf/bootstrap-4";
+import Alert from 'react-bootstrap/Alert'
 import RadioWidget from "./RadioWidget";
 import TextWidget from "./TextWidget";
 import styles from "./App.css";
 import "./Base.css";
 
-function updateConfig(formData) {
-  const config = formData.formData;
+function updateConfig(event, setProgress, setAlertVariant, setAlertText, setAlertVisibility, setButtonVisibility) {
+  // Saving state
+  setProgress("Saving...")
+  const config = event.formData;
+
   const response = fetch("/services/configurator/config", {
     method: "POST",
     headers: {
       "Content-Type": "application/json",
     },
     body: JSON.stringify(config),
+  })
+  .then(response => {
+    if (!response.ok) {
+      throw new Error('Form submission failed');
+    }
+    return response.status;
+  })
+  .then(status => {
+    // Submit successful, hide the alert
+    setAlertVariant("success");
+    setAlertVisibility(true);
+    setAlertText("Changes saved successfully!");
+    setProgress("Saved");
+    setButtonVisibility(false);
+  })
+  .catch(error => {
+    // Error on submit
+    setAlertText('There has been a problem with your form submit operation: \n' + error);
+    setAlertVariant('danger');
+    setProgress("Error");
   });
 }
 
+function onChange(setButtonVisibility, setAlertText, setAlertVariant, setAlertVisibility, setProgress) {
+  setAlertVisibility(true);
+  setButtonVisibility(true);
+  setProgress("Submit");
+  setAlertText("You have unsaved changes, click submit to save them.");
+  setAlertVariant("warning");
+}
```

Comment:
Having the parameters in a different order when compared with the body of the function rings something in my head... I know this is very minor but from a readability perspective, having the same ordering from left to right and from top to bottom could help the reader, IMIHO (this is also happening in the previous `updateConfig` function, I think).

Reply:
Great idea! Thanks @damianavila!